src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/TempPE/TempPEBuildManager.cs

```diff
@@ -0,0 +1,462 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+
+using Microsoft.Build.Execution;
+using Microsoft.VisualStudio.IO;
+using Microsoft.VisualStudio.LanguageServices.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.LanguageServices;
+using Microsoft.VisualStudio.ProjectSystem.VS.Automation;
+using Microsoft.VisualStudio.Threading.Tasks;
+
+using InputTuple = System.Tuple<Microsoft.VisualStudio.ProjectSystem.IProjectSnapshot, Microsoft.VisualStudio.ProjectSystem.IProjectSubscriptionUpdate>;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.TempPE
+{
+    [Export(typeof(ITempPEBuildManager))]
+    internal class TempPEBuildManager : UnconfiguredProjectHostBridge<IProjectVersionedValue<InputTuple>, DesignTimeInputsDelta, IProjectVersionedValue<DesignTimeInputsItem>>, ITempPEBuildManager
+    {
+        protected readonly IUnconfiguredProjectCommonServices _unconfiguredProjectServices;
+        private readonly ILanguageServiceHost _languageServiceHost;
+        private readonly IActiveConfiguredProjectSubscriptionService _projectSubscriptionService;
+        private readonly IFileSystem _fileSystem;
+        private readonly SequentialTaskExecutor _sequentialTaskQueue = new SequentialTaskExecutor();
+
+        // protected for unit test purposes
+        protected ITempPECompiler _compiler;
```

Comment:
Why? You pass the thing in the constructor why can't you use that?

Reply:
This just makes my test harness easier, so the test compiler can be given a reference to the test build manager.

I can rework the test infrastructure if this is important enough to worry about.